src/com/mitrais/atm/repository/AccountRepository.java

```diff
@@ -19,10 +25,36 @@ public static AccountRepository getInit() {
     public List<Account> loadData() {
         if (accountList == null) {
             accountList = new ArrayList<>();
-            accountList.add(new Account("John Doe", "112233", "012108", 100));
-            accountList.add(new Account("Jane Doe", "112244", "932012", 30));
         }
 
         return accountList;
     }
+
+    public void loadDataFile(String fileName) {
+        try (Stream<String> dataStream = Files.lines(Paths.get(fileName))) {
+            loadDataValidation(createAccountObject(dataStream));
+        } catch (IOException ioe) {
+            System.err.println("Error reading file: " + ioe);
+        }
+    }
+
+    private void loadDataValidation(Stream<Account> accountStream) {
+        AtomicBoolean keepAdding = new AtomicBoolean(true);
+        accountStream.peek(s -> keepAdding.set(AccountValidation.validate(s)))
+                .filter(s -> keepAdding.get())
+                .forEach(s -> loadData().add(s));
+    }
+
+    private Stream<Account> createAccountObject(Stream<String> dataStream) {
+        return dataStream.map(row -> {
+            String[] accountArr = row.split(";");
+
+            Account account = new Account();
+            account.setName(accountArr[0]);
+            account.setPin(accountArr[1]);
+            account.setBalance(Integer.valueOf(accountArr[2]));
```

Comment:
What happens if the value passed in the file is not parseable to Integer? Or if a row does not have all the specified positions? (4)

Reply:
I have adding the validator for checking per row 23e9d59